internal/formats/common/cyclonedxhelpers/hashes.go

```diff
@@ -0,0 +1,22 @@
+package cyclonedxhelpers
+
+import (
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/anchore/syft/syft/pkg"
+)
+
+func Hashes(p pkg.Package) *[]cyclonedx.Hash {
+	hashes := []cyclonedx.Hash{}
+	if hasMetadata(p) {
+		if metadata, ok := p.Metadata.(pkg.CargoPackageMetadata); ok && metadata.Checksum != "" {
```

Comment:
Today we surface checksum digest data from package metadata (in some cases). Additionally we have a file cataloger that observes the actual digests of the files from the source directly (container image or filesystem). Though it is convenient to report out the package metadata we should always report out the real filesystem hashes instead. We can cross correlate the filenames/paths from the package metadata and use that to access any digest information that the `sbom.SBOM` might have (the file digest cataloger is not enabled by default, so there may be no results).

Reply:
I don't think we should enable or disable specific catalogers conditionally based on which format is specified. I think that would start to get difficult to reason about. 

We should think about what catalogers we want in the default cataloger set in the future though. With the subcommand as `packages` enabling the digests cataloger doesn't make sense, but it could in the near future with other subcommands or deprecating the `packages` subcommand.

Reply:
We could definitely use cyclonedx file components to store the file level metadata (if it exists) and correlate that to the package component using bom refs in the future. I am happy to address it in a subsequent PR. This PR is getting a bit too big as is. We could only do it if the file level data/catalogers are enabled.

Reply:
agreed on not adding scope to this PR, however, I think the only remaining blocking comment (for me) is the use of hash information from package metadata. We should only be including hash information based on direct observation of the file contents and not trusting of the packaging metadata. 

We should remove this function from this PR for the meantime until we can include hashes sourced from the file itself (which is another feature/PR altogether).